db/sqldb.go

```diff
@@ -0,0 +1,557 @@
+		if err != nil {
+			return nil, err
+		}
+
+		// If there is a mountpoint, use it.
+		var mountpoint string
+		if mountpointNS.Valid {
+			mountpoint = mountpointNS.String
+		}
+
+		// Create volume
+		vol := volume.Volume{
+			Name:       name,
+			Mountpoint: mountpoint,
+			Status:     nil}
+
+		// Append it to the list
+		vols = append(vols, &vol)
+	}
+
+	// Check to see if there was an error durring interation
+	err = rows.Err()
+	if err != nil {
+		return nil, err
+	}
+
+	return vols, nil
+}
+
+// Get information about a particular volue
+func (s SQLVolumeDatabase) Get(volumeName string) (*volume.Volume, error) {
+	vol, _, err := s.getVolumeByName(volumeName)
+	return vol, err
+}
+
+func (s SQLVolumeDatabase) getVolumeIDByName(volumeName string) (int, error) {
+	_, id, err := s.getVolumeByName(volumeName)
+	if err == nil {
+		glog.Info(volumeName, " is id ", id)
+	}
+	return id, err
+}
+
+func (s SQLVolumeDatabase) getVolumeByName(volumeName string) (*volume.Volume, int, error) {
+	s.VerifyOrCrash()
+
+	// Prepare the query
+	stmt, err := sqlDB.Prepare(s.DBQueries.volumesGetVolumeByNameSQL)
+	if err != nil {
+		return nil, 0, err
+	}
+	defer stmt.Close()
+
+	// Query the database about the volumes
+	rows, err := stmt.Query(volumeName)
+	if err != nil {
+		return nil, 0, err
+	}
+	defer rows.Close()
+
+	// Itterate over all of the rows creating the slice of volumes we need. (there should only ever be one or zero)
+	var vols []*volume.Volume
+	var ids []int
+	for rows.Next() {
+		var id int
+		var name string
+		var mountpointNS sql.NullString
+		err = rows.Scan(&id, &name, &mountpointNS)
+		if err != nil {
+			return nil, 0, err
+		}
+
+		// If there is a mountpoint, use it.
+		var mountpoint string
+		if mountpointNS.Valid {
+			mountpoint = mountpointNS.String
+		}
+
+		// Create volume
+		vol := volume.Volume{
+			Name:       name,
+			Mountpoint: mountpoint,
+			Status:     nil}
+
+		// Append it to the list
+		vols = append(vols, &vol)
+		ids = append(ids, id)
+	}
+
+	// Check to see if there was an error durring interation
+	err = rows.Err()
+	if err != nil {
+		return nil, 0, err
+	}
+
+	// Did we get any results?
+	if len(vols) == 0 {
+		return nil, 0, errors.New("Volume does not exist.")
+	}
+
+	return vols[0], ids[0], nil
+}
+
+// Path returns the mountpath of a particular volume
+func (s SQLVolumeDatabase) Path(volumeName string) (string, error) {
+	vol, err := s.Get(volumeName)
+	if err != nil {
+		return "", err
+	}
+
+	return vol.Mountpoint, nil
+}
+
+// Remove (Delete) a volume from the database.
+func (s SQLVolumeDatabase) Remove(volumeName string) error {
+	_, requests, err := s.listMounts(volumeName)
+	if err != nil {
+		return err
+	}
+
+	if requests > 0 {
+		return errors.New("Volume cannot be removed as it still has active mount requests.")
+	}
+
+	id, err := s.getVolumeIDByName(volumeName)
+	if err != nil {
+		return err
+	}
+
+	// Begin transaction to the database
+	tx, err := sqlDB.Begin()
```

Comment:
Short variable names hinder the reader's ability to understand this code. Please lengthen them for all our sakes.